doc/lock_files.md

```diff
@@ -206,36 +177,8 @@ If the lock file does not exist, it will be created by:
 * Completing all missing information
 * Writing out the new `stack.yaml.lock` file
 
-## Dirtiness checking
-
-If the `stack.yaml.lock` file exists, its last modification time is
-compared against the last modification time of the `stack.yaml` file
-and any local snapshot files. If any of those files is more recent
-than the `stack.yaml` file, and the file hashes in the lock file
-do not match the files on the filesystem, then the update procedure is
-triggered. Otherwise, the `stack.yaml.lock` file can be used as the
-definition of the snapshot.
-
 ## Update procedure
 
-The simplest possible implementation is: ignore the lock file entirely
-and create a new one followign the creation steps above. There's a
-significant downside to this, however: it may cause a larger delta in
-the lock file than intended, by causing more packages to be
-updates. For example, many packages from Hackage may have their
-Hackage revision information updated unnecessarily.
-
-The more complicated update procedure is described below. **QUESTION**
-Do we want to go the easy way at first and later implement the more
-complicated update procedure?
-
-1. Create a map from original package location to completed package
-   location in the lock file
-2. Load up each snapshot file
-3. For each incomplete package location:
-    * Lookup the value in the map created in (1)
-    * If present: use that completed information
-    * Otherwise: complete the information using the same completion
-      procedure from Pantry as in "creation"
-
-This should minimize the number of changes to packages incurred.
+When loading a Stack project all completed package or snapshot locations
+(even when they were completed using information from a lock file) get
+collected to form a new lock file.
```

Comment:
> to form a new lock file

Can you clarify this a bit? I'm guessing the logic is "form a new lock file in memory and compare against the one on disk, writing if there are any differences." Is that the case?

Reply:
Yes, I'll add this

Reply:
done